client/components/ChallengesList.js

```diff
@@ -36,13 +37,30 @@ const component = ({
   const { searchTerm, difficulty } = match.params;
   let filter = {};
 
+  const challengesCollapseStatus = {};
+  const [collapseStatus, setCollapseStatus] = useState({});
+
+  const fetchCollapseStatus = arr => {
+    if (arr.length > 1) {
+      arr.forEach(challenge => {
+        challengesCollapseStatus[challenge.id] = false;
+      });
+    }
+  };
+
   useEffect(() => {
     fetchAllUserchallenges();
     if (difficulty) {
       filter = difficulty;
-      fetchChallengesWithFilterAndSearch(filter, searchTerm);
+      fetchChallengesWithFilterAndSearch(filter, searchTerm)
+        .then(resp => resp.challenges)
+        .then(challenges => fetchCollapseStatus(challenges))
+        .then(() => setCollapseStatus(challengesCollapseStatus));
```

Comment:
Let's catch the errors.  We will figure out how to handle them if we have time.
```suggestion
        .then(() => setCollapseStatus(challengesCollapseStatus))
        .catch(err => console.log(err));
```